### DISCUSSION TITLE
Use case: synchronization of an asynchronous master system communicating with a network of VSCP nodes

### DISCUSSION THREAD

**Discussion by michpro** [grodansparadis/vscp#276] [Q&A] [answered]
Hi Åke,

Please tell me how, using VSCP Level I messages, to handle the following case:
- modules communicate with each other, send appropriate messages (VSCP_TYPE_INFORMATION_ON, VSCP_TYPE_INFORMATION_OFF, etc.) received messages trigger actions - and it is OK - it works,
- now the problem: there is a master system, suppose “Home Assistant”, it crashes, or for some reason we have it restart. At this point, it does not remember the states that were sent in the network of VSCP nodes, so it is not synchronized with it. An event like turning off a light that is on may not occur until, say, several tens of minutes from now, and only then will synchronization occur. Solving this problem by periodically sending current states through all nodes will only introduce unnecessary noise on the bus, so this is not the best approach. I think that in such a case, our master system should be able to send, to the VSCP nodes, a message, requesting the current state (for example, resending all recent messages).... 

Maybe you have already included such a case in the protocol, but I can't find it, so please tell me how to solve it.

P.S. The solution with MQTT persistent messages, is also not advisable, it will work, but here it will only be a patch, not a real solution for VSCP network.

**Comment by grodansparadis** [comment, marked as answer]
Hi, as you write persistent messages/event over MQTT can be used as MQTT is one possible transport.  But the problem is the same there to if the broker go down. State changes while it is down does not benefit from persistent messages.   

Best IMHO is to read the current state from the (known) modules at startup on a node that need to know the last state of a node. Normally this is just a standard register read, and it is almost equally fast as reading some local stored values.  

Another method is to send (CLASS1-CONTROL, VSCP_TYPE_CONTROL_SYNC)[https://grodansparadis.github.io/vscp-doc-spec/#/./class1.control?id=type26] (or some other event of choice) to all nodes on start up and have them report there state. This can be programmed with the decision matrix of a device or be built in the nodes functionality. 

 (CLASS1-CONTROL, VSCP_TYPE_CONTROL_SYNC)[https://grodansparadis.github.io/vscp-doc-spec/#/./class1.control?id=type26] was added to synchronizing measurements in a scenario where you have many nodes that report measurements, and you need to get them as close in time as possible. It works equally well for states.

Personally, I use both methods.

**Reply by michpro** [reply]
Great, VSCP_TYPE_CONTROL_SYNC is what I need, and it looked so inconspicuous as I read the documentation :)
